### DISCUSSION TITLE
Deprecate MAKE and/or CLEAN, introduce BUILD_FLAGS

### DISCUSSION THREAD

**Discussion by evelikov** [dkms-project/dkms#321] [Polls]
Looking across various projects using dkms (bbswitch, nvidia driver, openzfs and a bunch of others). The vast majority of them copy/paste the default (in-dkms) make/clean commands:

```
make -C $kernel_source_dir M=$dkms_tree/$module/$module_version/build
make -C $kernel_source_dir M=$dkms_tree/$module/$module_version/build clean"
```

That has a few interesting side effects - both `kernel_source_dir` (aka the linux path) and `$dkms_tree` are hard-coded in their config.

Glancing at the more complex modules (nvidia, openzfs) - they are glorified wrappers allowing behaviour changes (via funky indirection).

Off the top of my head, to handle these we could introduce a variable which gets added to the make command above. Say `BUILD_FLAGS` or `BUILD_DEFINES` (must include only `-D...`) or alike. As always the `dkms.conf` file will have the default, which can be altered by users with the existing `conf.d` override mechanism.

@scaronni @xuzhen any preferences?

@hadogenes considering your experience with openzfs what your take?

**Comment by FlorianHeigl** [comment]
user here, i think the larger (vendor) modules aren't so relevant. they'll do their own thing primarily, what worries me is all the outdated / tiny stuff. i.e. when you fix/reactivate some forgotten broadcom 10g driver that actually works perfectly fine but noone took care of.
dkms is one of the few really reliable helps in such work.

**Comment by scaronni** [comment]
`CLEAN` has been deprecated in version 3.1.8. Along with all the other improvements done to the latest DKMS releases, I think we can leave MAKE as is right now.
